cloudflare/resource_cloudflare_teams_route.go

```diff
@@ -0,0 +1,144 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareTeamsRoute() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tunnel_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+		Create: resourceCloudflareTeamsRouteCreate,
+		Read:   resourceCloudflareTeamsRouteRead,
+		Update: resourceCloudflareTeamsRouteUpdate,
+		Delete: resourceCloudflareTeamsRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareTeamsRouteImport,
+		},
+	}
+}
+
+func resourceCloudflareTeamsRouteRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	tunnelRoute, err := client.GetTunnelRouteForIP(context.Background(), cloudflare.TunnelRoutesForIPParams{
+		AccountID: d.Get("account_id").(string),
+		Network:   d.Get("network").(string),
+	})
+	if err != nil {
+		// TODO: implement logic for missing route
```

Comment:
`d.SetId("")` will remove it from the state file if the response indicates the resource no longer exists. might need to check the response status or exception though. something like https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_access_rule.go#L102-L106 will work.